core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/GroupsAndRule.scala

```diff
@@ -18,20 +18,14 @@ package tech.beshu.ror.accesscontrol.blocks.rules.auth
 
 import tech.beshu.ror.accesscontrol.blocks.rules.Rule
 import tech.beshu.ror.accesscontrol.blocks.rules.Rule.{Name, RuleName}
-import tech.beshu.ror.accesscontrol.domain.{CaseSensitivity, Group, GroupsLogic, PermittedGroupIds}
-import tech.beshu.ror.utils.uniquelist.UniqueNonEmptyList
+import tech.beshu.ror.accesscontrol.domain.{CaseSensitivity, GroupsLogic}
 
-final class GroupsAndRule(override val settings: BaseGroupsRule.Settings,
+final class GroupsAndRule(override val settings: BaseGroupsRule.Settings[GroupsLogic.And],
```

Comment:
Please change the name to `Rule.Name("groups_all_of")` (it will be our default starting now). The "group_and" we will treat as a legacy name. 

Let's do the same for the GroupsOrRule

Reply:
I will be a part of task RORDEV-1388. It needs to be changed in multiple places (everywhere, where the current deprecated name of OR rule is used).